perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/MailManagerImpl.java

```diff
@@ -1388,351 +1433,401 @@ private String buildInviteURL(Vo vo, Group group, String text) {
+		mailText = replaceErrors(mailText, exceptions);
+		mailText = replaceLogins(mailText, app, data);
+		mailText = replaceAppDetailUrl(mailText, app.getId(), app.getVo(), app.getGroup());
+		mailText = replaceAppGuiUrl(mailText, app.getVo(), app.getGroup());
+		mailText = replacePerunGuiUrl(mailText, app.getVo(), app.getGroup());
+		mailText = replaceExpiration(mailText, app);
+		mailText = replaceMail(mailText, app, data);
+		mailText = replacePhone(mailText, app, data);
+
+		mailText = replaceFooter(FIELD_MAIL_FOOTER, app.getVo(), app.getGroup(), mailText, true);
+		mailText = replaceFooter(FIELD_HTML_MAIL_FOOTER, app.getVo(), app.getGroup(), mailText, isAlternativePlainText);
 
-		// replace app ID
-		if (mailText.contains(FIELD_APP_ID)) {
-			mailText = mailText.replace(FIELD_APP_ID, app.getId()+EMPTY_STRING);
+		return mailText;
+	}
+
+	private String replaceAppId(String mailText, Application app) {
+		if (!mailText.contains(FIELD_APP_ID)) {
+			return mailText;
 		}
+		return replaceInt(mailText, FIELD_APP_ID, app.getId());
+	}
 
-		// replace actor (app created by)
-		if (mailText.contains(FIELD_ACTOR)) {
-			mailText = mailText.replace(FIELD_ACTOR, app.getCreatedBy()+EMPTY_STRING);
+	private String replaceActor(String mailText, Application app) {
+		if (!mailText.contains(FIELD_ACTOR)) {
+			return mailText;
 		}
+		return replaceNullSafe(mailText, FIELD_ACTOR, app.getCreatedBy());
+	}
 
-		// replace ext source (app created by)
-		if (mailText.contains(FIELD_EXT_SOURCE)) {
-			mailText = mailText.replace(FIELD_EXT_SOURCE, app.getExtSourceName()+EMPTY_STRING);
+	private String replaceExtSource(String mailText, Application app) {
+		if (!mailText.contains(FIELD_EXT_SOURCE)) {
+			return mailText;
 		}
+		return replaceNullSafe(mailText, FIELD_EXT_SOURCE, app.getExtSourceName());
+	}
 
-		// replace voName
-		if (mailText.contains(FIELD_VO_NAME)) {
-			mailText = mailText.replace(FIELD_VO_NAME, app.getVo().getName());
+	private String replaceVoName(String mailText, Application app) {
+		if (!mailText.contains(FIELD_VO_NAME)) {
+			return mailText;
 		}
+		return replaceNullSafe(mailText, FIELD_VO_NAME, app.getVo().getName());
+	}
 
-		// replace groupName
-		if (mailText.contains(FIELD_GROUP_NAME)) {
-			if (app.getGroup() != null) {
-				mailText = mailText.replace(FIELD_GROUP_NAME, app.getGroup().getShortName());
-			} else {
-				mailText = mailText.replace(FIELD_GROUP_NAME, EMPTY_STRING);
-			}
+	private String replaceGroupName(String mailText, Application app) {
+		if (!mailText.contains(FIELD_GROUP_NAME)) {
+			return mailText;
+		}
+		String gName = EMPTY_STRING;
+		if (app.getGroup() != null) {
+			gName = app.getGroup().getShortName();
 		}
+		return replaceNullSafe(mailText, FIELD_GROUP_NAME, gName);
+	}
 
-		// replace customMessage (reason)
+	private String replaceCustomMessage(String mailText, String reason) {
 		if (mailText.contains(FIELD_CUSTOM_MESSAGE)) {
```

Comment:
```suggestion
		if (!mailText.contains(FIELD_CUSTOM_MESSAGE)) {
```
This triggers my OCD